presto-main/src/main/java/io/prestosql/sql/gen/InputReferenceCompiler.java

```diff
@@ -128,7 +128,11 @@ private InputReferenceNode(CallSiteBinder callSiteBinder, Scope scope, Type type
                     .pushJavaDefault(javaType);
 
             String methodName = "get" + Primitives.wrap(javaType).getSimpleName();
-            ifStatement.ifFalse(constantType(callSiteBinder, type).invoke(methodName, javaType, block, position));
+            BytecodeExpression value = constantType(callSiteBinder, type).invoke(methodName, javaType, block, position);
+            if (javaType == Object.class) {
```

Comment:
This looks weird. It’s not entirely obvious from looking at this code why it should be “==“. Either adjust the condition and how the information about whether to add the cast gets to this point (preferred), or add a comment to clarify.